Add canvas drawing functionality

GrandHENeral opened the pull request:
Gives users the ability to draw on a global canvas similar to r/place and Blurple event

GrandHENeral committed:
Add canvas drawing functionality.

    Similar to r/place or Blurple event
    The code now allows users to create and manipulate a canvas.
    Users can use the /place command to place pixels of different colors on the canvas once a minute.
    The /show command can be used to display the entire canvas or a zoomed-in portion of it.
    The /colorcodes command shows the available color codes for pixel placement.

GrandHENeral committed:
Code refactor

Mmesek commented on bot/events/place.py:
```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from mlib.graphing import create_image
```
This `create_image` import is not used, remove?

Mmesek commented on bot/events/place.py:
```diff
@@ -0,0 +1,156 @@
+    global canvas
+    try:
+        if os.path.isfile("canvas.json"):
+            with open("canvas.json", "r") as f:
```
All files of that kind are stored in `data/` directory. Perhaps prepend filenames with `data/`? 
Preferably also change using raw filename into a global constant variable too.

GrandHENeral committed:
Added docstring to colorcodes

Mmesek reviewed (changes requested):
Please clean up imports & Change file path and LGTM

GrandHENeral committed:
Code refactor
 -Removed unused imports
 -Made the json a global variable
 -All files get saved in data/ now

Mmesek commented on bot/events/place.py:
```diff
@@ -114,7 +116,7 @@ async def show(x: int = None, y: int = None, size: int = DEFAULT_ZOOM_SIZE):
 
     var = display_canvas(image)
-    return Attachment(file=var, filename="canvas.png")
+    return Attachment(file=var, filename="data/canvas.png")
```
That's a filename, you can't have path in a filename afaik

Mmesek commented on bot/events/place.py:
```diff
@@ -1,19 +1,21 @@
 # Default zoom when using the show command, will show a 50x50 pixel image
 DEFAULT_ZOOM_SIZE = 50
+#Path to canvas.json file
+canvas_file = "data/canvas.json"
```
Global constants are CAPITALIZED, you even have it capitalized in lines above. Capitalize for consistency?

GrandHENeral committed:
Code refactor based on review
